usd/src/usd_parser/USDTransforms.cc

```diff
@@ -0,0 +1,261 @@
+      if (!_tfs.empty()) {
+        auto& child = _tfs.back();
+        child.Pos().Set(
+          child.Pos().X() * t.scale[0],
+          child.Pos().Y() * t.scale[1],
+          child.Pos().Z() * t.scale[2]);
+      }
+
+      if (!t.isRotationZYX)
+      {
+        if (t.isRotation)
+        {
+          pose.Rot() = t.q[0];
+        }
+        _tfs.push_back(pose);
+      }
+      else
+      {
+        ignition::math::Pose3d poseZ = ignition::math::Pose3d(
+          ignition::math::Vector3d(0 ,0 ,0), t.q[2]);
+        ignition::math::Pose3d poseY = ignition::math::Pose3d(
+          ignition::math::Vector3d(0 ,0 ,0), t.q[1]);
+        ignition::math::Pose3d poseX = ignition::math::Pose3d(
+          ignition::math::Vector3d(0 ,0 ,0), t.q[0]);
+
+        ignition::math::Pose3d poseT = ignition::math::Pose3d(
+          t.translate * metersPerUnit,
+          ignition::math::Quaterniond(1, 0, 0, 0));
+
+        _tfs.push_back(poseZ);
+        _tfs.push_back(poseY);
+        _tfs.push_back(poseX);
+        _tfs.push_back(poseT);
+      }
+      parent = parent.GetParent();
+    }
+
+    if (upAxis == "Y")
+    {
+      ignition::math::Pose3d poseUpAxis = ignition::math::Pose3d(
+        ignition::math::Vector3d(0 ,0 ,0),
+        ignition::math::Quaterniond(IGN_PI_2, 0, 0));
+      _tfs.push_back(poseUpAxis);
+    }
+  }
+
+  void GetTransform(
+    const pxr::UsdPrim &_prim,
+    USDData &_usdData,
+    ignition::math::Pose3d &_pose,
+    ignition::math::Vector3d &_scale,
+    const std::string &_schemaToStop)
+  {
+    std::vector<ignition::math::Pose3d> tfs;
+    GetAllTransforms(_prim, _usdData, tfs, _scale, _schemaToStop);
+    for (auto & rt : tfs)
+    {
+      _pose = rt * _pose;
+    }
+  }
+
+  UDSTransforms ParseUSDTransform(const pxr::UsdPrim &_prim)
+  {
+    auto variant_geom = pxr::UsdGeomGprim(_prim);
+    auto transforms = variant_geom.GetXformOpOrderAttr();
+
+    pxr::GfVec3f scale(1, 1, 1);
+    pxr::GfVec3f translate(0, 0, 0);
+    pxr::GfQuatf rotationQuad(1, 0, 0, 0);
+
+    UDSTransforms t;
+
+    pxr::VtTokenArray xformOpOrder;
+    transforms.Get(&xformOpOrder);
+    for (auto & op: xformOpOrder)
+    {
+      if (op == "xformOp:scale")
+      {
+        auto attribute = _prim.GetAttribute(pxr::TfToken("xformOp:scale"));
+        if (attribute.GetTypeName().GetCPPTypeName() == "GfVec3f")
+        {
+          attribute.Get(&scale);
+        }
+        else if (attribute.GetTypeName().GetCPPTypeName() == "GfVec3d")
+        {
+          pxr::GfVec3d scaleTmp(1, 1, 1);
+          attribute.Get(&scaleTmp);
+          scale[0] = scaleTmp[0];
+          scale[1] = scaleTmp[1];
+          scale[2] = scaleTmp[2];
+        }
+        t.scale = ignition::math::Vector3d(scale[0], scale[1], scale[2]);
+      }
+      else if (op == "xformOp:rotateZYX" || op == "xformOp:rotateXYZ")
+      {
+        pxr::GfVec3f rotationEuler(0, 0, 0);
+        auto attribute = _prim.GetAttribute(pxr::TfToken("xformOp:rotateZYX"));
```

Comment:
This if block handles `rotateZYX` and `rotateXYZ`, but this only consider `rotateZYX`.